### PR TITLE
allow for multiple disk repairs

### PR DESCRIPTION
**Pull request by dr-bonez**

### CODE REVIEW

File: backend/src/disk/fsck.rs [hunk 1]
```diff
@@ -39,21 +41,38 @@ pub async fn e2fsck_preen(
     e2fsck_runner(Command::new("e2fsck").arg("-p"), logicalname).await
 }
 
+fn backup_existing_undo_file<'a>(path: &'a Path) -> BoxFuture<'a, Result<(), Error>> {
```
Comment (Blu-J) [inline backend/src/disk/fsck.rs:44]: Like the name is self explanatory, nice

Comment (Blu-J) [inline backend/src/disk/fsck.rs:44]: :question:  why didn't you make this fn async?

File: backend/src/disk/fsck.rs [hunk 2]
```diff
@@ -39,21 +41,38 @@ pub async fn e2fsck_preen(
...
+    async move {
+        if tokio::fs::metadata(path).await.is_ok() {
+            let bak = path.with_extension(format!(
+                "{}.bak",
+                path.extension()
+                    .and_then(|s| s.to_str())
+                    .unwrap_or_default()
+            ));
+            backup_existing_undo_file(&bak).await?;
```
Comment (Blu-J) [inline backend/src/disk/fsck.rs:53]: :speech_balloon: Nice that you have fixed the issue with already existing backup with a recursive method.

**Review by Blu-J — APPROVED** [review]

File: backend/src/disk/fsck.rs (hunk 1 above)
Comment (dr-bonez) [reply backend/src/disk/fsck.rs:45]: it is async, but recursive, so it needs to be boxed

Comment (Blu-J) [reply backend/src/disk/fsck.rs:45]: Glad I asked, now I learned something

**Pull request merged**